### PR TITLE
{ServiceBus} Add Breaking change message for `az servicebus namespace network-rule` cmdlets

### PR DESCRIPTION
**Pull request by schaudhari6254888**
**Related command**

**Description**
We will change the format of this command in Future release. the parameter `vnet-name` will be deprecated in future.
We will offer simultaneous addition and removal of multiple IP rules and subnet rules with the new future release cmdlets.
We are targeting these breaking change in this May breaking change cycle.
We will Replace the following command.

1. `az servicebus namespace network-rule` with `az servicebus namespace network-rule-set`

2. `az servicebus namespace network-rule add` with `az servicebus namespace network-rule-set ip-rule add`, `az servicebus namespace network-rule-set virtual-network-rule add`

3. `az servicebus namespace network-rule remove` with `az servicebus namespace network-rule-set ip-rule remove`, `az servicebus namespace network-rule-set virtual-network-rule remove`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by schaudhari6254888** [comment]
This PR is about breaking change for ServiceBus Network-Rule-Set cmdlets.
I have mentioned the changes in the description. Can someone review & merge this PR.
Any Suggestions are Welcome.
Thank you in advance!

File: src/azure-cli/azure/cli/command_modules/servicebus/custom.py [hunk 1]
```diff
@@ -229,6 +229,7 @@ def cli_networkrule_createupdate(cmd, client, resource_group_name, namespace_nam
     NWRuleSetIpRules = cmd.get_models('NWRuleSetIpRules', resource_type=ResourceType.MGMT_SERVICEBUS)
     netwrokruleset = client.get_network_rule_set(resource_group_name, namespace_name)
 
+    logger.warning('This version will be depracated & latest version will release in breaking change release.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/servicebus/custom.py:232]: You can leverage `deprecate_info` for specific command to show warnings and redirections

File: src/azure-cli/azure/cli/command_modules/servicebus/_params.py [hunk 2]
```diff
@@ -238,9 +238,9 @@ def load_arguments_sb(self, _):
     for scope in ['servicebus namespace network-rule add', 'servicebus namespace network-rule remove']:
         with self.argument_context(scope) as c:
             c.argument('subnet', arg_group='Virtual Network Rule', options_list=['--subnet'], help='Name or ID of subnet. If name is supplied, `--vnet-name` must be supplied.')
-            c.argument('ip_mask', arg_group='IP Address Rule', options_list=['--ip-address'], help='IPv4 address or CIDR range.')
+            c.argument('ip_mask', arg_group='IP Address Rule', options_list=['--ip-address'], help='IPv4 address or CIDR range.', deprecate_info=c.deprecate(redirect='--ip-rule', expiration='2.50.0'))
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/servicebus/_params.py:241]:
```suggestion
            c.argument('ip_mask', arg_group='IP Address Rule', options_list=['--ip-address'], help='IPv4 address or CIDR range.', deprecate_info=c.deprecate(redirect='--ip-rule', expiration='2.49.0'))
```
Pls double confirm the breaking change version

File: src/azure-cli/azure/cli/command_modules/servicebus/commands.py [hunk 3]
```diff
@@ -81,11 +81,13 @@ def load_command_table(self, _):
         g.custom_command('abort', 'revert')
 
 # NetwrokRuleSet Region
-    with self.command_group('servicebus namespace network-rule', sb_namespace_util, client_factory=namespaces_mgmt_client_factory, resource_type=ResourceType.MGMT_SERVICEBUS) as g:
-        g.custom_command('add', 'cli_networkrule_createupdate', validator=validate_subnet)
-        g.command('list', 'get_network_rule_set')
-        g.custom_command('remove', 'cli_networkrule_delete', validator=validate_subnet)
-        g.custom_command('update', 'cli_networkrule_update')
+    with self.command_group('servicebus namespace network-rule', sb_namespace_util, deprecate_info=self.deprecate(redirect='eventhubs namespace network-rule-set'), client_factory=namespaces_mgmt_client_factory, resource_type=ResourceType.MGMT_SERVICEBUS) as g:
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/servicebus/commands.py:84]: `servicebus namespace network-rule` will be redirected to `eventhubs namespace network-rule-set`??????
`servicebus` -> `eventhubs`???? You sure??????

Comment (schaudhari6254888) [reply src/azure-cli/azure/cli/command_modules/servicebus/commands.py]: Sorry, My Bad.
Resolved.

File: src/azure-cli/azure/cli/command_modules/servicebus/_params.py (hunk 2 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/servicebus/_params.py]: Last release was `2.47.0`............

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**